### PR TITLE
feat(cloudfront): add grantCreateInvalidation to Distribution

### LINKED ISSUE #13159: (aws-cloudfront): Add convenience grant methods
Hey hey,

ideally the cloudfront distribution would provide convenience `.grant*` methods, to e.g. grant a user the permissions to invalidate a distribution or get a invalidation so the can wait on it until completion.

### Use Case

A common use case is creating a CI/CD user with permissions to
- publish assets to a distribution's origin like a s3 bucket
- followed by invalidating the cloudfront distribution

for example this could be a CI/CD user automatically building and publishing static assets.

Now it would be great to have functionality like

```js
const dist = new Distribution(..);
const user = new User(...);

dist.grantCreateInvalidation(user);
```

or similar, instead of having to write the policy out manually.

### PR DESCRIPTION
**Pull request by Tietew**
This PR adds `distributionArn` attribute and `grantCreateInvalidation()` method to `IDistribution`.

- `distributionArn` - returns the ARN of the distribution.
- `grantCreateInvalidation()` - grants `cloudfront:CreateInvalidation` on the distribution.

Fixes #13159

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cloudfront/lib/distribution.ts [hunk 1]
```diff
@@ -39,6 +41,20 @@ export interface IDistribution extends IResource {
    * @attribute
    */
   readonly distributionId: string;
+
+  /**
+   * The distribution ARN for this distribution.
+   *
+   * @attribute
+   */
+  readonly distributionArn: string;
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-cloudfront/lib/distribution.ts:50]: I'm not sure I get why this is being added in this context. It's not included in `Fn::GetAtt` in the resource per https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudfront-distribution.html. How is this being used?

**Review by TheRealAmazonKendra — DISMISSED** [review]
Thank you for your contribution! Please see my inline comments.

File: packages/@aws-cdk/aws-cloudfront/lib/private/utils.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IDistribution } from '..';
+
+/**
+ * Format distribution ARN from stack and distribution ID.
+ */
+export function formatDistributionArn(scope: Construct, distributionId: string) {
```
[the remark this answers is not in the document]
Comment (Tietew) [reply packages/@aws-cdk/aws-cloudfront/lib/private/utils.ts:9]: These functions are shared by `distribution.ts` and `web-distribution.ts`.

File: packages/@aws-cdk/aws-cloudfront/lib/distribution.ts (hunk 1 above)
Comment (Tietew) [reply packages/@aws-cdk/aws-cloudfront/lib/distribution.ts:50]: It's useful when grant other than `CreateInvalidation`:

```ts
declare const role: iam.Role;

// before
role.addToPrincipalPolicy(new iam.PolicyStatement({
  actions: ['cloudfront:ListInvalidations'],
  // region is needed here
  resources: [this.formatArn({ service: 'cloudfront', region: '', resource: 'distribution', resourceName: dist.distributionId })],
}));

// after
role.addToPrincipalPolicy(new iam.PolicyStatement({
  actions: ['cloudfront:ListInvalidations'],
  resources: [dist.distributionArn],
}));
```

Can be removed if it's excessive.

**Comment by Tietew** [comment]
@TheRealAmazonKendra I created an alternative PR #22709. Would you please choose which PR is approvable?
Thank you.

**Pull request closed without merging**